### PR TITLE
RavenDB-18228 - Sharding - Edit External Replication View Endpoints

### PR DESCRIPTION
**Pull request by shiranshalom**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18228

### Type of change

- New feature

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing 

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks
+{
+    internal abstract class AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication<TRequestHandler> : AbstractHandlerProcessorForUpdateDatabaseConfiguration<BlittableJsonReaderObject, TRequestHandler>
+        where TRequestHandler : RequestHandler
+    {
+        protected AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication.cs:24]: you forgot to implement FillJson (from the old code). IIRC this is called now OnBeforeConfigurationUpdate

**Review by ppekrol — COMMENTED** [review]
Please also check failing tests

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication.cs (hunk 1 above)
Comment (shiranshalom) [reply src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForUpdateExternalReplication.cs:24]: done

**Review by karmeli87 — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
@Danielle9897 please rebase and resolve conflict

**Comment by ml054** [comment]
conflict resolved

**Pull request merged**